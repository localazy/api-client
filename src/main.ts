export * from '~/api/api-client';
export * from '~/api/methods/api-base';
export * from '~/api/methods/api-export';
export * from '~/api/methods/api-files';
export * from '~/api/methods/api-formats';
export * from '~/api/methods/api-glossary';
export * from '~/api/methods/api-import';
export * from '~/api/methods/api-keys';
export * from '~/api/methods/api-projects';
export * from '~/api/methods/api-screenshots';
export * from '~/api/methods/api-webhooks';
export * from '~/enums/i18n-deprecate';
export * from '~/enums/project-tone';
export * from '~/enums/project-type';
export * from '~/enums/user-role';
export * from '~/enums/webhook-event';
export * from '~/errors/localazy-error';
export * from '~/http/axios-http-adapter';
export * from '~/http/i-http-adapter';
export * from '~/types/api-client-options';
export * from '~/types/export-json-request';
export * from '~/types/file';
export * from '~/types/file-get-contents-request';
export * from '~/types/file-list-keys-request';
export * from '~/types/files-list-request';
export * from '~/types/format';
export * from '~/types/format-array';
export * from '~/types/format-array-meta';
export * from '~/types/format-key-transformer-meta';
export * from '~/types/format-plural';
export * from '~/types/format-plural-meta';
export * from '~/types/format-plural-meta-required-param';
export * from '~/types/glossary-create-request';
export * from '~/types/glossary-delete-request';
export * from '~/types/glossary-find-request';
export * from '~/types/glossary-list-request';
export * from '~/types/glossary-record';
export * from '~/types/glossary-record-term';
export * from '~/types/glossary-update-request';
export * from '~/types/i18n-json';
export * from '~/types/import-content-options';
export * from '~/types/import-data';
export * from '~/types/import-data-factory';
export * from '~/types/import-data-file';
export * from '~/types/import-file-options';
export * from '~/types/import-i18n-options';
export * from '~/types/import-json-request';
export * from '~/types/json';
export * from '~/types/key';
export * from '~/types/key-delete-request';
export * from '~/types/key-update-request';
export * from '~/types/key-value';
export * from '~/types/keys-paginated';
export * from '~/types/language';
export * from '~/types/locales-keys';
export * from '~/types/organization';
export * from '~/types/project';
export * from '~/types/projects-list-request';
export * from '~/types/screenshot';
export * from '~/types/screenshot-create-request';
export * from '~/types/screenshot-delete-request';
export * from '~/types/screenshot-metadata';
export * from '~/types/screenshot-phrase';
export * from '~/types/screenshot-tag';
export * from '~/types/screenshot-update-image-data-request';
export * from '~/types/screenshot-update-request';
export * from '~/types/screenshots-list-request';
export * from '~/types/screenshots-list-tags-request';
export * from '~/types/webhook';
export * from '~/types/webhooks-get-secret-request';
export * from '~/types/webhooks-list-request';
export * from '~/types/webhooks-secret';
export * from '~/types/webhooks-update-request';
export * from '~/utils/json-utils';
export { Locales } from '@localazy/languages';
