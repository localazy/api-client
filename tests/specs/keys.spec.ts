import { beforeEach, describe, expect, test } from 'vitest';
import { getApiClient, mockAdapter } from '@tests/support';
import { fullProject } from '@tests/fixtures';
import { ApiClient, File, Key, KeyDeleteRequest, KeyUpdateRequest, Locales, Project } from '@/main';

describe('Keys', (): void => {
  let api: ApiClient;
  let project: Project;

  beforeEach(async (): Promise<void> => {
    fullProject.mockResponses();

    api = getApiClient();
    project = await api.projects.first();
  });

  test('api.keys.update', async (): Promise<void> => {
    const file: File = await api.files.first({ project });
    const keys: Key[] = await api.files.listKeys({ project, file, lang: Locales.ENGLISH });
    const request: KeyUpdateRequest = { project, key: keys[0], comment: 'Comment for translators.' };
    await api.keys.update(request);

    expect(mockAdapter.history.put.length).toBe(1);
    expect(mockAdapter.history.put[0].data).toMatchSnapshot();
  });

  test('api.keys.delete', async (): Promise<void> => {
    const file: File = await api.files.first({ project });
    const keys: Key[] = await api.files.listKeys({ project, file, lang: Locales.ENGLISH });
    const request: KeyDeleteRequest = { project, key: keys[0] };
    await api.keys.delete(request);

    expect(mockAdapter.history.delete.length).toBe(1);
    expect(mockAdapter.history.delete[0].data).toMatchSnapshot();
  });
});
