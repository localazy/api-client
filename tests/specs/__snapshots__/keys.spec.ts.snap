// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Keys > api.keys.delete 1`] = `undefined`;

exports[`Keys > api.keys.update 1`] = `"{"comment":"Comment for translators."}"`;
